nbclient/util.py

```diff
@@ -28,6 +30,7 @@ def check_patch_tornado() -> None:
             tornado.concurrent.FUTURES = \
                 tornado.concurrent.FUTURES + (asyncio.Future, )  # type: ignore
 
+pool = concurrent.futures.ThreadPoolExecutor()
```

Comment:
Any reason to use more than one thread? I think we should use `max_workers=1`.

Reply:
max_worker should be fine IMO.

Reply:
If you don't specify `mar_workers`, it will use [unnecessary resources](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor): "If max_workers is None or not given, it will default to the number of processors on the machine, multiplied by 5".
Since you never actually run concurrently, you don't need more than 1 thread.